### PR TITLE
JBPM-4646 - Case management - added PerCase runtime strategy and runt…

### PR DESCRIPTION
**Pull request by mswiderski**
…ime manager

### CODE REVIEW

**Comment by mswiderski** [comment]
@cristianonicolai @krisv first version ready to be reviewed and merged

File: kie-internal/src/main/java/org/kie/internal/runtime/manager/context/CaseContext.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2013 JBoss Inc
```
Comment (cristianonicolai) [inline kie-internal/src/main/java/org/kie/internal/runtime/manager/context/CaseContext.java]: You've been hiding this code for 3 years? :P

**Comment by cristianonicolai** [comment]
@mswiderski looks good to me apart from the minor issue with the license header.

File: kie-internal/src/main/java/org/kie/internal/runtime/manager/context/CaseContext.java (hunk 1 above)
Comment (mswiderski) [inline kie-internal/src/main/java/org/kie/internal/runtime/manager/context/CaseContext.java]: updated header

P.S.
yeah, I have plenty of these changes to "release to public" that I worked on them for years... ;)

File: kie-internal/src/main/java/org/kie/internal/runtime/manager/context/CaseContext.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2016 JBoss Inc
```
Comment (cristianonicolai) [inline kie-internal/src/main/java/org/kie/internal/runtime/manager/context/CaseContext.java]: Red Hat, Inc. and/or its affiliates.

File: kie-api/src/main/java/org/kie/api/runtime/manager/RuntimeManagerFactory.java [hunk 3]
```diff
@@ -87,6 +87,24 @@
     RuntimeManager newPerProcessInstanceRuntimeManager(RuntimeEnvironment environment, String identifier);
 
     /**
+     * Produces new instance of per case <code>RuntimeManager</code> with default identifier.
+     * Since it relies on default identifier it can only be invoked once unless previously produced
+     * manager is closed. Otherwise error will be thrown indicating that managers must be identifier uniquely.
+     * @param environment environment instance for the new runtime manager
+     * @return
```
Comment (rsynek) [inline kie-api/src/main/java/org/kie/api/runtime/manager/RuntimeManagerFactory.java]: return shouldn't be empty - javadoc will complain about that.

**Comment by mswiderski** [comment]
@cristianonicolai @rsynek changes applied, thanks

**Comment by cristianonicolai** [comment]
+1

**Pull request merged**